Tasks/Gradle/CodeAnalysis/SonarQube/common.ts

```diff
@@ -172,52 +71,108 @@ function isPrBuild(): boolean {
         sourceBranch.toLowerCase().startsWith("refs/pull/");
 }
 
+/**
+ * Returns, as an object, the contents of the report-task.txt file created by SonarQube plugins
+ * The returned object contains the following properties:
+ *     projectKey, serverUrl, dashboardUrl, ceTaskId, ceTaskUrl
+ * @param sonarPluginFolder The folder created by SonarQube integration during build
+ * @returns {SonarQubeRunSettings} An object with fields corresponding to the properties exposed in report-task.txt, or null if PR build
+ */
+function getSonarQubeRunSettings(sonarPluginFolder: string): SonarQubeRunSettings {
+    if (isPrBuild()) {
+        return null;
```

Comment:
Returning null is a deliberate decision here. The report-task.txt file does not exist in PR builds (it is the only scenario where we expect to not find it), but there may still be some integration actions that we can do without it.